Tests/Marketplace/marketplace_services.py

```diff
@@ -2365,6 +2373,182 @@ def is_unified_integration(self, file_path: str):
+        """
+        Receives changelog, checks if there exists a BC version in each changelog entry (as changelog entry might be
+        zipped into few RN versions, check if at least one of the versions is BC).
+        Check if RN is BC is done by doing the following:
+         1) Check if RN has corresponding config file, e.g 1_0_1.md has corresponding 1_0_1.json file.
+         2) If it does, check if `isBreakingChanges` field is true
+        If such version exists, adds a
+        true value to 'breakingChanges' field.
+        if JSON file also has breakingChangesNotes configures, adds `breakingChangesNotes` field to changelog file.
+        This function iterates every entry in changelog because it takes into consideration four scenarios:
+          a) Entry without breaking changes, changes to entry with breaking changes (because at least one of the
+             versions in the entry was marked as breaking changes).
+          b) Entry without breaking changes, does not change.
+          c) Entry with breaking changes, changes to entry without breaking changes (because all the BC versions
+             corresponding to the changelog entry were re-marked as not BC).
+          d) Entry with breaking changes, does not change.
+        Args:
+            release_notes_dir (str): RN dir path.
+            changelog (Dict[str, Any]): Changelog data represented as a dict.
+
+        Returns:
+            (None): Modifies changelog, adds bool value to 'breakingChanges' and `breakingChangesNotes` fields to every
+             changelog entry, according to the logic described above.
+        """
+        if not os.path.exists(release_notes_dir):
+            return
+        bc_version_to_text: Dict[str, Optional[str]] = self._breaking_changes_versions_to_text(release_notes_dir)
+        loose_versions: List[LooseVersion] = [LooseVersion(bc_ver) for bc_ver in bc_version_to_text.keys()]
+        predecessor_version: LooseVersion = LooseVersion('0.0.0')
+        for changelog_entry in sorted(changelog.keys(), key=LooseVersion):
+            rn_loose_version: LooseVersion = LooseVersion(changelog_entry)
+            if bc_versions := self._changelog_entry_bc_versions(predecessor_version, rn_loose_version, loose_versions,
+                                                                bc_version_to_text):
+                logging.info(f'Changelog entry {changelog_entry} contains BC versions')
+                changelog[changelog_entry]['breakingChanges'] = True
+                if bc_text := self._calculate_bc_text(release_notes_dir, bc_versions):
+                    changelog[changelog_entry]['breakingChangesNotes'] = bc_text
+                else:
+                    changelog[changelog_entry].pop('breakingChangesNotes', None)
+            else:
+                logging.info('Version no BC!')
+                changelog[changelog_entry].pop('breakingChanges', None)
```

Comment:
```suggestion
                changelog[changelog_entry].pop('breakingChanges')
```

Reply:
Are you sure this required? Why should the changelog have a 'breakingChanges' key if it didn't have any bc versions? (thats the `if` condition to this `else`)

Reply:
If some version was BC but then someone decided to modify it to be a non-BC version